desc/objectives/objective_funs.py

```diff
@@ -742,6 +747,15 @@ def __init__(
         assert normalize_target in {True, False}
         assert (bounds is None) or (isinstance(bounds, tuple) and len(bounds) == 2)
         assert (bounds is None) or (target is None), "Cannot use both bounds and target"
+        assert callable(loss_function), "Loss function must be callable!"
+        assert (
+            len(getfullargspec(loss_function).args) == 1
+        ), "Loss function must accept a single array as an argument!"
+        _test_output_loss = loss_function(jnp.array([1, 1]))
```

Comment:
might be better to pass an array of size `dim_f` here and then update the `dim_f` attribute afterwards with the correct output size.